ResearchKit/Common/ORKSurveyAnswerCellForText.m

```diff
@@ -327,12 +351,12 @@ - (BOOL)textField:(UITextField *)textField shouldChangeCharactersInRange:(NSRang
 }
 
 - (BOOL)textFieldShouldEndEditing:(UITextField *)textField {
-    return YES;
+    return [self isAnswerValidForTextField:textField];
 }
 
 - (BOOL)textFieldShouldReturn:(UITextField *)textField {
     [self.textField resignFirstResponder];
-    return YES;
+    return [self isAnswerValidForTextField:textField];
```

Comment:
Same as above. Let's revert this back to `return YES` to stay consistent with our existing user experience.